kimchi/js/bindings.js

```diff
@@ -2199,11 +2226,43 @@ var caml_pasta_fp_commitments_to_rust = function (x) {
   var w_comm = convertArray(x[1]);
   var z_comm = caml_vesta_poly_comm_to_rust(x[2]);
   var t_comm = caml_vesta_poly_comm_to_rust(x[3]);
-  return new plonk_wasm.WasmFpProverCommitments(w_comm, z_comm, t_comm);
+  var lookup;
+  if (x[4] === 0) {
+    lookup = undefined;
+  } else {
+    lookup = caml_fp_lookup_commitments_to_rust(x[4][1]);
+  }
```

Comment:
I think it would be clearer if we could write
```
if (is_none(x[4)) { ...} else { ... = f(get_option(x[4]) }
```
Maybe for another issue ?

Reply:
I totally agree with this. As it would cover other parts of the code, I think opening an issue, and link with https://github.com/o1-labs/snarkyjs-bindings/pull/110#discussion_r1295583790 is better.